src/main/scala/io/iohk/ethereum/jsonrpc/PersonalService.scala

```diff
@@ -157,7 +157,12 @@ class PersonalService(
       val maybeCurrentNonce = getCurrentAccount(request.from).map(_.nonce.toBigInt)
       val maybeNextTxNonce = maybeLatestPendingTxNonce.map(_ + 1) orElse maybeCurrentNonce
       val tx = request.toTransaction(maybeNextTxNonce.getOrElse(blockchainConfig.accountStartNonce))
-      val stx = wallet.signTx(tx)
+
+      val stx = if(blockchain.getBlockByNumber(blockchainConfig.eip155BlockNumber).isDefined){
+        wallet.signTx(tx, Some(blockchainConfig.chainId))
+      }else{
+        wallet.signTx(tx, None)
+      }
```

Comment:
Well, not exactly. While the current method is perhaps not a problem, it makes more sense to me to do `appStateStorage.getBestBlockNumber() +1 >= blockchainConfig.eip155BlockNumber` (it may also be faster?). And that, I think, would be more similar to what geth does.